ramanujan/matrix.py

```diff
@@ -40,29 +40,42 @@ def simplify(self):
         """Returns a simplified version of matrix"""
         return Matrix(sp.simplify(self))
 
-    def walk(self, trajectory, iterations, start):
+    def walk(self, trajectory, iterations, start, scoops = set()):
```

Comment:
IMO you should merge the `iterations` and `scoops` logic into one. Receive a list of sorted depths, return the matrix for each one of them.